website/client/js/services/userServices.js

```diff
@@ -103,8 +104,27 @@ angular.module('habitrpg')
 
           return Tasks.getUserTasks();
         })
+        // refresh all but completed todos
         .then(function (response) {
-          var tasks = response.data.data;
+          tasks.push.apply(tasks, response.data.data);
```

Comment:
in this case you have an empty array before adding the fetched data, just replace it